src/platform/macos/view.rs

```diff
@@ -280,7 +280,7 @@ extern fn insert_text(this: &Object, _sel: Sel, string: id, _replacement_range:
             characters.len(),
         );
         let string = str::from_utf8_unchecked(slice);
-        state.last_insert = Some(string.to_owned());
+        state.is_key_down = true;
```

Comment:
I don't remember the specifics, I'm take a look when I'm back at my computer. Thanks for looking, and thanks for your hard work on winit!

Reply:
@francesca64 Ok, so, `last_insert` was only used in in the `is_repeat` block, maybe as a way to avoid the logic I've added. The problem is `last_insert` didn't catch changes in modifiers.

Reply:
Alright, sounds good! Thanks for the PR!